db/migrate/20200430205827_deconcatenate_subject_in_item_and_thesis.rb

```diff
@@ -4,7 +4,11 @@ def change
     add_column :items, :deconcatenated_subject, :json, array: true
     add_column :theses, :deconcatenated_subject, :json, array: true
 
-    @concatenated_subject_ids = File.readlines(Rails.root.join('ERA_subject_issue.txt')).collect(&:strip)
+    @concatenated_subject_ids = if File.exist?(Rails.root.join('ERA_subject_issue.txt'))
+      File.readlines(Rails.root.join('ERA_subject_issue.txt')).collect(&:strip)
+    else
+      []
+    end
```

Comment:
I was considering using a guard clause (see suggestion) but I think that in order for this migration to make sense the `deconcatenated_subject` column needs to be filled.
```suggestion
    return unless File.exist?(Rails.root.join('ERA_subject_issue.txt'))
    @concatenated_subject_ids = File.readlines(Rails.root.join('ERA_subject_issue.txt')).collect(&:strip)
```